lib/index.js

```diff
@@ -290,24 +430,37 @@ api.get = async ({actor, account, email, type} = {}) => {
  * @param {string} options.account - The ID of the account to remove a token
  *   from.
  * @param {string} options.type - The type of token to remove.
+ * @param {string} [options.id] - The id of the token to be removed.
  *
  * @returns {Promise} - A Promise that resolves once the operation completes.
  */
-api.remove = async ({actor, account, type} = {}) => {
+api.remove = async ({actor, account, type, id} = {}) => {
   assert.string(account, 'account');
+  assert.optionalString(id, 'id');
   validateTokenType(type);
 
   await brPermissionCheck(
     actor, PERMISSIONS.ACCOUNT_UPDATE, {resource: [account]});
-
+  let update;
+  const tokenTypeKey = `meta.bedrock-authn-token.tokens.${type}`;
+  if(type === 'nonce') {
```

Comment:
We need to ensure that `id` is defined when the type is `nonce`, otherwise the query below will include `undefined`.